ReactCommon/fabric/components/text/paragraph/ParagraphShadowNode.cpp

```diff
@@ -50,11 +50,15 @@ void ParagraphShadowNode::updateLocalDataIfNeeded() {
   }
 
   auto localData = std::make_shared<ParagraphLocalData>();
-  localData->setAttributedString(std::move(attributedString));
+  // localData->setAttributedString(std::move(attributedString));
```

Comment:
The biggest problem is here:
You cannot create a State inside ShadowNode. It already should be created *before* creation of ShadowNode. And it's already done automatically if your ShadowNode has defined StateData.

You have to call `getData().setAttributedString(...)`. And the biggest problem here is that `setAttributedString` must be `const`. So you have to ensure inside this method that it is being called only once (check some boolean inside (we need a boolean because AttributedString does not have "undefined" value)).

Reply:
And we also need to assign LayoutManager here. OR we can assign it inside static method `ConcreteShadowNode::initialStateData` (which is better).

Reply:
Can you clarify:
```
And we also need to assign LayoutManager here. OR we can assign it inside static method ConcreteShadowNode::initialStateData (which is better).
```

I am not really sure how to use initialStateData, and I *think* I am assigning it to state, but it doesn't seem to work.